File: Math/976. Largest Perimeter Triangle.cpp
class Solution {
public:
    int largestPerimeter(vector<int>& nums) 
    {
        int ans = 0, n = nums.size();
        sort(begin(nums), end(nums));
        for(int i = 1; i < n - 1; i++)
        {
            int a = nums[i - 1], b = nums[i], c = nums[i + 1];
            if(a + b > c)
                ans = max(ans, (a + b + c));
        }
        return ans;    
    }
};

File: 976. Largest Perimeter Triangle.py
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort()
        nums.reverse()
        
        n = len(nums)
        for i in range(1, n - 1):
            a, b, c = nums[i + 1], nums[i], nums[i - 1]
            if a + b > c:
                return a + b + c
        return 0